staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -199,7 +193,18 @@ func (s *Scheme) AddKnownTypeWithName(gvk schema.GroupVersionKind, obj Object) {
 		panic("All types must be pointers to structs.")
 	}
 
+	if oldT, found := s.gvkToType[gvk]; found && oldT != t {
```

Comment:
Preserving stack traces is only for debugging, won't be merged.

Am not so sure about panicking on duplicated registration. We might have situation where it's tricky to avoid repeated registration, compare https://github.com/kubernetes/kubernetes/pull/41757/files#diff-a32a22caf2988f380979f0fefbcdce13R28.